cql3/expr/expression.hh

```diff
@@ -532,6 +553,28 @@ nested_expression::nested_expression(ExpressionElement auto e)
 // For example "WHERE c = 1 AND (a, c) = (2, 1) AND token(p) < 2 AND FALSE" will return {"c = 1"}.
 std::vector<expression> extract_single_column_restrictions_for_column(const expression&, const column_definition&);
 
+std::optional<bool> get_bool_value(const constant&);
+
+// Takes a prepared expression and calculates its value.
+// Later term will be replaced with expression.
+constant evaluate(const ::shared_ptr<term>&, const query_options&);
+constant evaluate(term*, const query_options&);
+constant evaluate(term&, const query_options&);
+
+// Calls evaluate() on the term and then converts the constant to raw_value_view
+cql3::raw_value_view evaluate_to_raw_view(const ::shared_ptr<term>&, const query_options&);
+cql3::raw_value_view evaluate_to_raw_view(term&, const query_options&);
+
+utils::chunked_vector<managed_bytes> get_list_elements(const constant&);
+utils::chunked_vector<managed_bytes> get_set_elements(const constant&);
+std::vector<managed_bytes_opt> get_tuple_elements(const constant&);
+std::vector<managed_bytes_opt> get_user_type_elements(const constant&);
+std::vector<std::pair<managed_bytes, managed_bytes>> get_map_elements(const constant&);
+std::vector<managed_bytes_opt> get_elements(const constant&);
```

Comment:
I'd add a blurb to the last one, at least.

Reply:
Added